src/sentry/static/sentry/app/components/createAlertButton.tsx

```diff
@@ -0,0 +1,266 @@
+  const {
+    hasProjectError,
+    hasEnvironmentError,
+    hasEventTypeError,
+    hasYAxisError,
+  } = incompatibleQuery;
+
+  const totalErrors = Object.values(incompatibleQuery).filter(val => val === true).length;
+
+  return (
+    <StyledAlert type="warning" icon={<IconInfo color="yellow400" size="sm" />}>
+      {totalErrors === 1 && (
+        <React.Fragment>
+          {hasProjectError && t('Select one project to create a new alert.')}
+          {hasEnvironmentError &&
+            t('Select one or all environments to create a new alert.')}
+          {hasEventTypeError &&
+            tct(
+              'Select either [error:event.type:error] or [transaction:event.type:transaction] to create a new alert.',
+              {
+                error: <StyledCode />,
+                transaction: <StyledCode />,
+              }
+            )}
+          {hasYAxisError &&
+            tct(
+              '[yAxis] is not supported by alerts just yet. Select a different metric below and try again.',
+              {
+                yAxis: <StyledCode>{eventView.getYAxis()}</StyledCode>,
+              }
+            )}
+        </React.Fragment>
+      )}
+      {totalErrors > 1 && (
+        <React.Fragment>
+          {t(
+            'The world is a cruel and unforgiving place and that button didn’t work because:'
+          )}
+          <StyledUnorderedList>
+            {hasProjectError && <li>{t('One project must be selected')}</li>}
+            {hasEnvironmentError && <li>{t('One or All Environments is required')}</li>}
+            {hasEventTypeError && (
+              <li>
+                {tct(
+                  'Either [error:event.type:error] or [transaction:event.type:transaction] is required',
+                  {
+                    error: <StyledCode />,
+                    transaction: <StyledCode />,
+                  }
+                )}
+              </li>
+            )}
+            {hasYAxisError && (
+              <li>
+                {tct(
+                  '[yAxis] is not supported by alerts just yet. Select a different metric below and try again.',
+                  {
+                    yAxis: <StyledCode>{eventView.getYAxis()}</StyledCode>,
+                  }
+                )}
+              </li>
+            )}
+          </StyledUnorderedList>
+        </React.Fragment>
+      )}
+      <StyledCloseButton
+        icon={<IconClose color="yellow400" size="sm" isCircled />}
+        aria-label={t('Close')}
+        size="zero"
+        onClick={onClose}
+        borderless
+      />
+    </StyledAlert>
+  );
+}
+
+type Props = React.ComponentProps<typeof Button> & {
+  className?: string;
+  projects: Project[];
+  /**
+   * Discover query used to create the alert
+   */
+  eventView: EventView;
+  organization: Organization;
+  /**
+   * Called when the eventView does not meet the requirements of alert rules
+   */
+  onIncompatibleQuery: (incompatibleQuery: React.ReactNode) => void;
```

Comment:
variable name should probably be `alert`, even though it's mostly unimportant in types